tools/systemd/generate-pgroonga-primary-maintainer-service.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+set -xue
+
+pgroonga_primary_maintainer_command=$(which pgroonga-primary-maintainer.sh || :)
+threshold="1G"
+psql_command=""
+environments=()
+on_failure_service=""
+
+function usage () {
+  cat <<USAGE
+Options:
+-s, --pgroonga-primary-maintainer-command:
+  Specify the path to \`pgroonga-primary-maintainer.sh\`
+  (default: ${pgroonga_primary_maintainer_command})
+-t, --threshold:
+  If the specified value is exceeded, \`REINDEX INDEX CONCURRENTLY\` is run.
+  (default: ${threshold})
+-e, --environment
+  Connection information such as \`dbname\` should be set in environment variables.
+  See also: https://www.postgresql.org/docs/current/libpq-envars.html"
+  Example: --environment KEY1=VALUE1 --environment KEY2=VALUE2 ...
+-c, --psql:
+  Specify the path to \`psql\` command.
+-h, --help:
+  Display help text and exit.
+USAGE
+}
+
+short_options="s:t:e:c:h"
+long_options="pgroonga_primary_maintainer_command:,threshold:,environment:,psql:,help"
+# If you run `getopt` with no arguments and get an error,
+# you are in an environment where the `--longoptions` option is available.
+if getopt > /dev/null; then
```

Comment:
Can we always use GNU `getopt` because systemd is used only on Linux for now?

Reply:
Since we assume Linux only, use GNU `getopt`.